docs/source/library-user-guide/api-health.md

```diff
@@ -69,3 +69,32 @@ For example:
 Deprecated methods will remain in the codebase for a period of 6 major versions or 6 months, whichever is longer, to provide users ample time to transition away from them.
 
 Please refer to [DataFusion releases](https://crates.io/crates/datafusion/versions) to plan ahead API migration
+
+## Migration Guidelines
+
+To ensure smooth upgrades and maintain application stability, the following guidelines must be followed for changes involving:
```

Comment:
Stylistically I would prefer to say this more gently "the following guidelines should be followeed" rather than must. Mostly because there is no way for us to enforce this policy other than pull request reviews

However, I think that may just be a cultural hangup I have as I come from the US and the distinction may not be relevant for other contributors

Reply:
fixed